ext/dcos-installer/dcos_installer/action_lib/configure.py

```diff
@@ -6,34 +7,45 @@
 import gen
 import pkgpanda
 import gen.installer.bash
-from dcos_installer.util import SERVE_DIR
+from dcos_installer.util import ARTIFACT_DIR, GENCONF_DIR, SERVE_DIR
 
 log = logging.getLogger(__name__)
 
 
 def do_configure(gen_config):
+    # Generate complete config.
     gen_config.update(get_gen_extra_args())
-
-    subprocess.check_output(['mkdir', '-p', SERVE_DIR])
-
     do_validate_gen_config(gen_config)
-
     gen_out = gen.generate(arguments=gen_config)
-    gen.installer.bash.generate(gen_out, SERVE_DIR)
 
-    # Get bootstrap from artifacts
-    fetch_bootstrap(gen_out.arguments['bootstrap_id'])
-    # Write some package metadata
-    pkgpanda.util.write_json('/genconf/cluster_packages.json', gen_out.cluster_packages)
+    # Generate install/upgrade prereqs.
+    subprocess.check_output(['mkdir', '-p', SERVE_DIR])
+    gen.installer.bash.generate(gen_out, SERVE_DIR)
+    fetch_bootstrap(ARTIFACT_DIR, SERVE_DIR, gen_out.arguments['bootstrap_id'])
+    fetch_packages(ARTIFACT_DIR, SERVE_DIR)
+    pkgpanda.util.write_json(os.path.join(GENCONF_DIR, 'cluster_packages.json'), gen_out.cluster_packages)
 
 
 def get_gen_extra_args():
     if 'BOOTSTRAP_ID' not in os.environ:
         log.error("BOOTSTRAP_ID must be set in environment to run.")
         raise KeyError
 
+    # Get package IDs for this build from build metadata.
+    try:
+        package_ids = pkgpanda.util.load_json(os.path.join(ARTIFACT_DIR, 'complete.json'))['packages']
+    except IOError as exc:
+        message = "Can't find build metadata in artifacts."
+        log.error(message)
+        raise Exception(message) from exc
+    except ValueError as exc:
+        message = "Can't read build metadata."
+        log.error(message)
+        raise Exception(message) from exc
+
     arguments = {
         'bootstrap_id': os.environ['BOOTSTRAP_ID'],
+        'package_ids': json.dumps(package_ids),
```

Comment:
this isn't your fault, but this isn't how we should be doing this / where we should be doing this.  Should be adding it to: https://github.com/dcos/dcos/blob/master/gen/calc.py#L407, so that it applies across providers, since in theory all of them want / need this full set of package ids.